app/validators/linked_validator.rb

```diff
@@ -0,0 +1,50 @@
+class LinkedValidator < ActiveModel::EachValidator
+  DELIMITER = ':'.freeze
+
+  def validate_each(record, attribute, value)
+    register_linked = options[:register_linked]
+    record.errors.add attribute, "#{value} is not a valid CURIE" unless validate_linked(value, register_linked)
```

Comment:
The error message suggested by UCD for this case is *Must be valid data from a register that is ready to use* 
I do not really like this but we should use it for now.